handlers/authHandlers.js

```diff
@@ -13,6 +16,17 @@ var EXPIRY = process.env.EXPIRY || '1d';
 var DEFAULT_USER_TYPE = process.env.DEFAULT_USER_TYPE || 'Null';
 var PUBKEY;
 var PRIKEY;
+var GENERATE_KEY_IF_MISSING = process.env.GENERATE_KEY_IF_MISSING || true;
+
+if (!fs.existsSync('./keys/key') && !fs.existsSync('./keys/key.pub') && GENERATE_KEY_IF_MISSING) {
+  try {
+    exec(`${preCommand}'/CN=www.camicroscope.com/O=caMicroscope Local Instance Key./C=US'${postCommand}`);
+    console.log('keys are generated. please restart the server.' + '\n');
+    process.exit(0);
```

Comment:
What happens if the server doesn't exit here? Is there a benefit to it exiting?

Reply:
all the validation is handled by the server. if the server doesn't exist  `authHandler.js` never run so no required for key validation. correct me if I am wrong.

Reply:
I mean, you create the key, so the next steps would be able to find key and key.pub if the process kept running, right?
I think you may need to use execSync though.

Reply:
sure. i made the changes below.